_posts/2020-09-07-introduction-to-system-tests.md

```diff
@@ -0,0 +1,275 @@
+    String modelStr = TestUtils.toYamlString(model);  
+    assertDoesNotThrow(() -> createDelete(modelStr), 
+    "Create delete failed after first round-trip -- maybe a problem with a defaulted value?\nApplied string: " + modelStr);
+}
+```
+
+You can see that the test case basically applies the YAML representation of Kafka and expects the creation of a `Kafka` customer 
+resource object in Kubernetes cluster is correct.
+This is applied for every custom resource supported by `Strimzi`, such as `KafkaConnect`, `KafkaMirrorMaker`, `KafkaBridge`,
+`KafkaTopic`, `KafkaUser` and so on.
+
+### System tests
+
+After this level is successfully completed we can talk about our system tests.
+Take a look at the informal definition again:
+
+> System testing is a level of testing that validates the complete and fully integrated software product.
+> The purpose of a system test is to evaluate the end-to-end system specifications.
+> Usually, the software is only one element of a larger computer-based system.
+> Ultimately, the software is interfaced with other software/hardware systems.
+> System Testing is actually a series of different tests whose sole purpose is to exercise the full computer-based system. (Guru)
+
+In our case the system tests validates all components and features which that Strimzi provides.
+Everything is tested in a Kubernetes environment to fully represent a production or user environment.
+Again, to clarify I have an example:
+
+```java
+@Test
+void testReceiveSimpleMessage() {
+    KafkaTopicResource.topic(CLUSTER_NAME, TOPIC_NAME).done();
+
+    kafkaBridgeClientJob.consumerStrimziBridge().done();
+
+    // Send messages to Kafka
+    InternalKafkaClient internalKafkaClient = new InternalKafkaClient.Builder()
+        .withTopicName(TOPIC_NAME)
+        .withNamespaceName(NAMESPACE)
+        .withClusterName(CLUSTER_NAME)
+        .withMessageCount(MESSAGE_COUNT)
+        .withKafkaUsername(USER_NAME)
+        .withUsingPodName(kafkaClientsPodName)
+        .build();
+
+    assertThat(internalKafkaClient.sendMessagesPlain(), is(MESSAGE_COUNT));
+
+    ClientUtils.waitForClientSuccess(consumerName, NAMESPACE, MESSAGE_COUNT);
+}
+
+@BeforeAll
+void createClassResources() throws Exception {
+    deployClusterOperator(NAMESPACE);
+    LOGGER.info("Deploy Kafka and KafkaBridge before tests");
+    // Deploy kafka
+    KafkaResource.kafkaEphemeral(CLUSTER_NAME, 1, 1).done();
+
+    KafkaClientsResource.deployKafkaClients(false, KAFKA_CLIENTS_NAME).done();
+
+    kafkaClientsPodName = kubeClient().listPodsByPrefixInName(KAFKA_CLIENTS_NAME).get(0).getMetadata().getName();
+
+    // Deploy http bridge
+    KafkaBridgeResource.kafkaBridge(CLUSTER_NAME, KafkaResources.plainBootstrapAddress(CLUSTER_NAME), 1)
+        .editSpec()
+            .withNewConsumer()
+                .addToConfig(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest")
+            .endConsumer()
+        .endSpec()
+        .done();
```

Comment:
Again, some comments in this code would help the reader understand what's going on here.